neuraxle/steps/flow.py

```diff
@@ -0,0 +1,43 @@
+from neuraxle.base import MetaStepMixin, BaseStep, ExecutionContext, DataContainer, NonTransformableMixin, \
+    ExecutionMode, NonFittableMixin
+
+
+class TransformOnlyWrapper(NonTransformableMixin, NonFittableMixin, MetaStepMixin, BaseStep):
+    """
+    A wrapper step that only executes in the transform execution mode.
+
+    .. seealso:: :class:`ExecutionMode`, :class:`DataContainer`, :class:`NonTransformableMixin`, :class:`NonFittableMixin`, :class:`MetaStepMixin`, :class:`BaseStep`
+    """
+
+    def handle_transform(self, data_container: DataContainer, context: ExecutionContext) -> DataContainer:
+        return self.wrapped.handle_transform(data_container, context)
+
+
+class FitTransformOnlyWrapper(NonTransformableMixin, NonFittableMixin, MetaStepMixin, BaseStep):
+    """
+    A wrapper step that only executes in the fit_transform execution mode.
+
+    .. seealso:: :class:`ExecutionMode`, :class:`DataContainer`, :class:`NonTransformableMixin`, :class:`NonFittableMixin`, :class:`MetaStepMixin`, :class:`BaseStep`
+    """
+
+    def handle_fit_transform(self, data_container: DataContainer, context: ExecutionContext):
+        self.wrapped, outputs = self.wrapped.handle_fit_transform(data_container, context)
+        return self, outputs
+
+
+class FitOnlyWrapper(NonTransformableMixin, NonFittableMixin, MetaStepMixin, BaseStep):
```

Comment:
Is inheriting from `NonFittableMixin` dangerous or borderline here?

well it seems to compile and work, perhaps it just feels less intuitive.

Reply:
I would advise against inheriting from the non-fittable mixin in a fit-only wrapper. Here is a rare case where I'd prefer rewriting the needed methods manually instead of inheriting.

Reply:
no, I only want to call the wrappe methods. closing this. look at the new code, and reopen if needed.